frontend/src/actions/transfers/transfer.ts

```diff
@@ -103,6 +118,36 @@ export class Transfer extends MultiStepAction implements Encodable<TransferData>
     return super.executeSteps(methods);
   }
 
+  public async withdraw(provider: IEthereumProvider): Promise<void> {
+    if (!this._expired) {
+      throw new Error('Can only withdraw transfer funds after request expired!');
+    }
+
+    // Theoretically impossible given the above condition.
+    if (!this._requestInformation?.identifier) {
+      throw new Error('Attempting to withdraw without request identifier!');
+    }
+
+    if (!provider.signer.value) {
+      throw new Error('Can not withdraw without connected wallet!');
+    }
+
+    const currentChainIdentifier = await provider.getChainId();
+
+    if (currentChainIdentifier !== this.sourceChain.identifier) {
+      provider.switchChain(this.sourceChain.identifier);
```

Comment:
Does the page really reload? I think in the other places the reload is manually triggered after the `switchChain` call. We could also change the provider implementation to do it there.

Reply:
We observe the chain identifier and reload it [here](https://github.com/beamer-bridge/beamer/blob/2f251fae7cebfc328b6ceabaeecaf0f7fa3eacac/frontend/src/components/RequestDialog.vue#L126).
I kinda don't like this whole page reloading on chain change etc. I'm not 100% sure if it is actually necessary at all. So I guess I tried to minimize this effect application.
Anyway, I think you have a good point and I can ensure that this happens at the `EthereumProvider`. There it is really close to the source of event + change and no Vue component has to take care of that.

Reply:
> There was some error that made us introduce the location reload thing. I could find the related PR: #388

I also read in the [MetaMask](https://docs.metamask.io/guide/ethereum-provider.html#chainchanged) docs that it is strongly recommended. I still would love to know the exact reasoning. 
Thank you very much or the link!

Reply:
I think I'm mixing up two things here. Reload when we change the source chain and reloading when the wallet changed the network.
The first part I want to change in future. The second is already solved properly.
Thereby I'm closing. 👍🏾